src/opengl/adapters.cpp

```diff
@@ -429,6 +434,8 @@ class GlslFragmentShaderImplementation : public GlslFragmentShader {
     {
         glGenBuffers(1, &vertexBuffer);
         glGenBuffers(1, &vertexIndexBuffer);
+        glGenVertexArrays(1, &vao);
+        glBindVertexArray(vao);
```

Comment:
Without `glBindVertexArray(vao);` here, tests will crash. I don't understand why though, because it's still done in bind(). I suspect this might only work in scenario with single program and fail when there are multiple.

Reply:
Makes sense if we need a VAO to be able to `glBindBuffer(GL_ARRAY_BUFFER, ...)`. I guess it's similar to the relationship between `glBindFrameBuffer` and `glFramebufferTexture2D`. I had no idea that the VAO was a thing that needed to be handled.

For consistency, I guess it would be better to also call `glBindVertexArray(0)` at the end of this ctor too

Reply:
Yes, but then it doesn't work and I don't understand why :)

It will crash on the draw call with invalid operation error. That's why I'm a bit concerned about this piece of code